### DISCUSSION TITLE
Setting different confidence thresholds for different classes

### DISCUSSION THREAD

**Discussion by jeannotdamoiseaux** [ultralytics/yolov5#7746] [Q&A] [answered]
Is it already possible to set different confidence threshold values for different classes? I haven't been able to find this functionality. This would be very valuable for my application, where raising the confidence threshold may be beneficial for one class, but detrimental for another.

**Comment by glenn-jocher** [comment, marked as answer]
@jeannot-github this is an interesting idea, but there's no feature implemented currently for this. val.py does return metrics per class, so you could conceivably use these to determine a best confidence threshold per class, i.e. perhaps at the maximum F1 confidence for each class for the best real-world P and R balance:

![image](https://user-images.githubusercontent.com/26833433/167603457-a0848b6a-35fd-4ac8-9183-538b9ed9e5af.png)

**Comment by sonovice** [comment]
Such a feature would be especially useful for multi class boxes where the softmax output is (best case scenario ->) mostly distributed uniformly among x correct classes, resulting in lower scores. So multi class and single class detections need different thresholds to be reliable.
